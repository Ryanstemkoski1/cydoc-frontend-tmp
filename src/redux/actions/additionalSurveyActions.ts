import { UPDATE_ADDITIONAL_DETAILS } from './actionTypes';

export interface UpdateAdditionalSurveyAction {
    type: UPDATE_ADDITIONAL_DETAILS.UPDATE_ADDITIONAL_DETAILS;
    payload: {
        legalFirstName: string;
        legalLastName: string;
        socialSecurityNumber: string;
        dateOfBirth: string;
        showAdditionalSurvey: boolean;
    };
}

export interface GoBackToAdditionalSurvey {
    type: UPDATE_ADDITIONAL_DETAILS.GO_BACK;
    payload: {
        showAdditionalSurvey: boolean;
    };
}

export function updateAdditionalSurveyDetails(
    legalFirstName: string,
    legalLastName: string,
    socialSecurityNumber: string,
    dateOfBirth: string
): UpdateAdditionalSurveyAction {
    return {
        type: UPDATE_ADDITIONAL_DETAILS.UPDATE_ADDITIONAL_DETAILS,
        payload: {
            legalFirstName,
            legalLastName,
            socialSecurityNumber,
            dateOfBirth,
            showAdditionalSurvey: false,
        },
    };
}

export function resetAdditionalSurveyPage(): GoBackToAdditionalSurvey {
    return {
        type: UPDATE_ADDITIONAL_DETAILS.GO_BACK,
        payload: {
            showAdditionalSurvey: true,
        },
    };
}
