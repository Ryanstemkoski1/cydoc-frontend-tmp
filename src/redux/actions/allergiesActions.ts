import { ALLERGIES_ACTION } from './actionTypes';

interface HasAllergiesAction {
    type: ALLERGIES_ACTION.HAS_ALLERGIES;
    payload: {
        hasAllergies: boolean;
    };
}

export function toggleHasAllergies(state: boolean): HasAllergiesAction {
    return {
        type: ALLERGIES_ACTION.HAS_ALLERGIES,
        payload: {
            hasAllergies: state,
        },
    };
}

interface UpdateIncitingAgentAction {
    type: ALLERGIES_ACTION.UPDATE_INCITING_AGENT;
    payload: {
        index: string;
        newIncitingAgent: string;
    };
}

export function updateIncitingAgent(index: string, newIncitingAgent: string) {
    return {
        type: ALLERGIES_ACTION.UPDATE_INCITING_AGENT,
        payload: {
            index,
            newIncitingAgent,
        },
    };
}

interface UpdateReactionAction {
    type: ALLERGIES_ACTION.UPDATE_REACTION;
    payload: {
        index: string;
        newReaction: string;
    };
}

export function updateReaction(index: string, newReaction: string) {
    return {
        type: ALLERGIES_ACTION.UPDATE_REACTION,
        payload: {
            index,
            newReaction,
        },
    };
}

interface UpdateCommentsAction {
    type: ALLERGIES_ACTION.UPDATE_COMMENTS;
    payload: {
        index: string;
        newComments: string;
    };
}

export function updateComments(index: string, newComments: string) {
    return {
        type: ALLERGIES_ACTION.UPDATE_COMMENTS,
        payload: {
            index,
            newComments,
        },
    };
}

interface UpdateIdAction {
    type: ALLERGIES_ACTION.UPDATE_ID;
    payload: {
        index: string;
        id: string;
    };
}

export function updateId(index: string, id: string) {
    return {
        type: ALLERGIES_ACTION.UPDATE_ID,
        payload: {
            index,
            id,
        },
    };
}

interface AddAllergyAction {
    type: ALLERGIES_ACTION.ADD_ALLERGY;
}

export function addAllergy() {
    return {
        type: ALLERGIES_ACTION.ADD_ALLERGY,
    };
}

interface DeleteAllergyAction {
    type: ALLERGIES_ACTION.DELETE_ALLERGY;
    payload: {
        index: string;
    };
}

export function deleteAllergy(index: string) {
    return {
        type: ALLERGIES_ACTION.DELETE_ALLERGY,
        payload: {
            index,
        },
    };
}

export type AllergiesActionTypes =
    | HasAllergiesAction
    | UpdateIncitingAgentAction
    | UpdateReactionAction
    | UpdateCommentsAction
    | UpdateIdAction
    | AddAllergyAction
    | DeleteAllergyAction;
