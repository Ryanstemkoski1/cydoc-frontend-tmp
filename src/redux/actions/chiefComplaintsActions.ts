import { CHIEF_COMPLAINTS } from './actionTypes';

export interface SelectChiefComplaintAction {
    type: CHIEF_COMPLAINTS.SELECT_CHIEF_COMPLAINTS;
    payload: {
        disease: string;
    };
}

export function selectChiefComplaint(
    disease: string
): SelectChiefComplaintAction {
    return {
        type: CHIEF_COMPLAINTS.SELECT_CHIEF_COMPLAINTS,
        payload: {
            disease,
        },
    };
}

export type chiefComplaintsActionTypes = SelectChiefComplaintAction;
