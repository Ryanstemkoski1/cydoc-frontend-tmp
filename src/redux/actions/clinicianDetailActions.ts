import { initialClinicianDetailType } from 'redux/reducers/clinicianDetailReducer';
import { CLINICIAN_DETAILS } from './actionTypes';

export interface clinicianDetailAction {
    type: CLINICIAN_DETAILS;
    payload: {
        [key: string]: string | number | initialClinicianDetailType;
    };
}

export function setClinicianLastName(last_name: string) {
    return {
        type: CLINICIAN_DETAILS.SET_CLINICIAN_LAST_NAME,
        payload: {
            last_name,
        },
    };
}

export function setClinicianFirstName(first_name: string) {
    return {
        type: CLINICIAN_DETAILS.SET_CLINICIAN_FIRST_NAME,
        payload: {
            first_name,
        },
    };
}

export function setClinicianEmail(email: string) {
    return {
        type: CLINICIAN_DETAILS.SET_CLINICIAN_EMAIL,
        payload: {
            email,
        },
    };
}

export function setClinicianId(id: number) {
    return {
        type: CLINICIAN_DETAILS.SET_CLINICIAN_ID,
        payload: {
            id,
        },
    };
}

export function setClinicianInstitutionId(institution_id: number) {
    return {
        type: CLINICIAN_DETAILS.SET_CLINICIAN_ID,
        payload: {
            institution_id,
        },
    };
}
export function setClinicianDetail(obj: initialClinicianDetailType) {
    return {
        type: CLINICIAN_DETAILS.SET_CLINICIAN_DETAIL,
        payload: {
            obj,
        },
    };
}
