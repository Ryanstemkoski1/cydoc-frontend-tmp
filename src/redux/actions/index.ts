import { ROSActionTypes } from './reviewOfSystemsActions';
import { MedicalHistoryActionTypes } from './medicalHistoryActions';
import { SurgicalHistoryActionTypes } from './surgicalHistoryActions';
import { MedicationsActionTypes } from './medicationsActions';
import { AllergiesActionTypes } from './allergiesActions';
import { SocialHistoryActionTypes } from './socialHistoryActions';
import { FamilyHistoryActionTypes } from './familyHistoryActions';
import { PhysicalExamActionTypes } from './physicalExamActions';
import { PlanActionTypes } from './planActions';
import { HpiActionTypes } from './hpiActions';
import { WidgetActionTypes } from './widgetActions';
import { chiefComplaintsActionTypes } from './chiefComplaintsActions';

export type AllActionTypes =
    | ROSActionTypes
    | PhysicalExamActionTypes
    | WidgetActionTypes
    | MedicalHistoryActionTypes
    | SurgicalHistoryActionTypes
    | MedicationsActionTypes
    | AllergiesActionTypes
    | SocialHistoryActionTypes
    | FamilyHistoryActionTypes
    | PlanActionTypes
    | HpiActionTypes
    | chiefComplaintsActionTypes;
