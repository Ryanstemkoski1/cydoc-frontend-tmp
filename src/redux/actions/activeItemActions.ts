import { UnknownAction } from 'redux';
import { ACTIVE_ITEM_ACTION } from './actionTypes';

export interface UpdateActiveItemAction extends UnknownAction {
    type: ACTIVE_ITEM_ACTION.ACTIVE_ITEM;
    payload: {
        updatedItem: string;
    };
}

export function updateActiveItem(updatedItem: string): UpdateActiveItemAction {
    return {
        type: ACTIVE_ITEM_ACTION.ACTIVE_ITEM,
        payload: {
            updatedItem,
        },
    };
}

export type activeItemActionTypes = UpdateActiveItemAction;
