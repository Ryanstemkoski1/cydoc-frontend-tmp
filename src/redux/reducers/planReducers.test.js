import { planReducer, initialPlanState } from './planReducer';
import { PLAN_ACTION } from '@redux/actions/actionTypes';
import { WhenResponse, YesNoUncertainResponse } from 'constants/enums';
import {
    initialPlan,
    conditionId,
    categoryId,
} from 'pages/EditNote/content/discussionplan/util';

describe('plan reducers', () => {
    describe('general condition reducers', () => {
        it('returns the initial state', () => {
            expect(planReducer(undefined, {})).toEqual(initialPlanState);
        });

        it('adds new condition', () => {
            const newConditions = planReducer(initialPlan, {
                type: PLAN_ACTION.ADD_CONDITION,
            }).conditions;

            expect(Object.keys(newConditions).length).toEqual(2);
            expect(conditionId in newConditions).toBeTruthy();
        });

        it('deletes initial condition', () => {
            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.DELETE_CONDITION,
                    payload: { conditionIndex: conditionId },
                })
            ).toMatchSnapshot();
        });

        it('updates initial condition name', () => {
            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.UPDATE_CONDITION_NAME,
                    payload: {
                        conditionIndex: conditionId,
                        newName: 'foo',
                    },
                })
            ).toMatchSnapshot();
        });
    });

    describe('differential diagnoses', () => {
        let payload = {};
        beforeEach(() => {
            payload = {
                conditionIndex: conditionId,
                diagnosisIndex: categoryId,
            };
        });

        it('adds new differential diagnosis', () => {
            const newDifferentialDiagnoses = planReducer(initialPlan, {
                type: PLAN_ACTION.ADD_DIFFERENTIAL_DIAGNOSIS,
                payload,
            }).conditions[conditionId].differentialDiagnoses;

            expect(Object.keys(newDifferentialDiagnoses).length).toEqual(2);
            expect(categoryId in newDifferentialDiagnoses).toBeTruthy();
        });

        it('updates diagnosis', () => {
            payload.newDiagnosis = 'foo';

            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.UPDATE_DIFFERENTIAL_DIAGNOSIS,
                    payload,
                })
            ).toMatchSnapshot();
        });

        it('updates comments', () => {
            payload.newComments = 'foo';

            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.UPDATE_DIFFERENTIAL_DIAGNOSIS_COMMENTS,
                    payload,
                })
            ).toMatchSnapshot();
        });

        it('deletes differential diagnoses', () => {
            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.DELETE_DIFFERENTIAL_DIAGNOSIS,
                    payload,
                })
            ).toMatchSnapshot();
        });
    });

    describe('prescriptions', () => {
        let payload = {};
        beforeEach(() => {
            payload = {
                conditionIndex: conditionId,
                prescriptionIndex: categoryId,
            };
        });

        it('adds new prescription', () => {
            const newPrescriptions = planReducer(initialPlan, {
                type: PLAN_ACTION.ADD_PRESCRIPTION,
                payload,
            }).conditions[conditionId].prescriptions;

            expect(Object.keys(newPrescriptions).length).toEqual(2);
            expect(categoryId in newPrescriptions).toBeTruthy();
        });

        it('updates prescription comments', () => {
            payload.newComments = 'foo';

            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.UPDATE_PRESCRIPTION_COMMENTS,
                    payload,
                })
            ).toMatchSnapshot();
        });

        it('updates prescription dose', () => {
            payload.newDose = 'foo';

            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.UPDATE_PRESCRIPTION_DOSE,
                    payload,
                })
            ).toMatchSnapshot();
        });

        it('updates prescription type', () => {
            payload.newType = 'foo';

            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.UPDATE_PRESCRIPTION_TYPE,
                    payload,
                })
            ).toMatchSnapshot();
        });

        it('updates prescription signature', () => {
            payload.newSignature = 'foo';

            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.UPDATE_PRESCRIPTION_SIGNATURE,
                    payload,
                })
            ).toMatchSnapshot();
        });

        it('deletes prescription', () => {
            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.DELETE_PRESCRIPTION,
                    payload,
                })
            ).toMatchSnapshot();
        });
    });

    describe('procedure or service', () => {
        let payload = {};
        beforeEach(() => {
            payload = {
                conditionIndex: conditionId,
                procedureIndex: categoryId,
            };
        });

        it('adds new procedure or service', () => {
            const newProcedures = planReducer(initialPlan, {
                type: PLAN_ACTION.ADD_PROCEDURE_OR_SERVICE,
                payload,
            }).conditions[conditionId].proceduresAndServices;

            expect(Object.keys(newProcedures).length).toEqual(2);
            expect(categoryId in newProcedures).toBeTruthy();
        });

        it('updates procedure', () => {
            payload.newProcedure = 'foo';
            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.UPDATE_PROCEDURE_OR_SERVICE,
                    payload,
                })
            ).toMatchSnapshot();
        });

        it('updates comments', () => {
            payload.newComments = 'foo';
            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.UPDATE_PROCEDURE_OR_SERVICE_COMMENTS,
                    payload,
                })
            ).toMatchSnapshot();
        });

        it('updates when', () => {
            payload.newWhen = WhenResponse.ThisMonth;
            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.UPDATE_PROCEDURE_OR_SERVICE_WHEN,
                    payload,
                })
            ).toMatchSnapshot();
        });

        it('deletes procedure or services', () => {
            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.DELETE_PROCEDURE_OR_SERVICE,
                    payload,
                })
            ).toMatchSnapshot();
        });
    });

    describe('referrals', () => {
        let payload = {};
        beforeEach(() => {
            payload = {
                conditionIndex: conditionId,
                referralIndex: categoryId,
            };
        });

        it('adds new procedure or service', () => {
            const newReferrals = planReducer(initialPlan, {
                type: PLAN_ACTION.ADD_REFERRAL,
                payload,
            }).conditions[conditionId].referrals;

            expect(Object.keys(newReferrals).length).toEqual(2);
            expect(categoryId in newReferrals).toBeTruthy();
        });

        it('updates department', () => {
            payload.newDepartment = 'foo';
            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.UPDATE_REFERRAL_DEPARTMENT,
                    payload,
                })
            ).toMatchSnapshot();
        });

        it('updates comments', () => {
            payload.newComments = 'foo';
            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.UPDATE_REFERRAL_COMMENTS,
                    payload,
                })
            ).toMatchSnapshot();
        });

        it('updates when', () => {
            payload.newWhen = WhenResponse.ThisMonth;
            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.UPDATE_REFERRAL_WHEN,
                    payload,
                })
            ).toMatchSnapshot();
        });

        it('deletes referral', () => {
            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.DELETE_REFERRAL,
                    payload,
                })
            ).toMatchSnapshot();
        });
    });

    describe('survey reducers', () => {
        it('updates admit to hospital', () => {
            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.UPDATE_ADMIT_TO_HOSPITAL,
                    payload: {
                        newAdmitToHospital: YesNoUncertainResponse.Yes,
                    },
                })
            ).toMatchSnapshot();
        });

        it('updates emergency', () => {
            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.UPDATE_EMERGENCY,
                    payload: {
                        newEmergency: YesNoUncertainResponse.No,
                    },
                })
            ).toMatchSnapshot();
        });

        it('updates sickness', () => {
            expect(
                planReducer(initialPlan, {
                    type: PLAN_ACTION.UPDATE_SICKNESS_LEVEL,
                    payload: {
                        newSicknessLevel: 5,
                    },
                })
            ).toMatchSnapshot();
        });
    });
});
