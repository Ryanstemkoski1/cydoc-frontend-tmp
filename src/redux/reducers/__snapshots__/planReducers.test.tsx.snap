// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`plan reducers > differential diagnoses > deletes differential diagnoses 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {},
      "name": "",
      "prescriptions": {
        "category": {
          "comments": "",
          "dose": "",
          "signature": "",
          "type": "",
        },
      },
      "proceduresAndServices": {
        "category": {
          "comments": "",
          "procedure": "",
          "when": "",
        },
      },
      "referrals": {
        "category": {
          "comments": "",
          "department": "",
          "when": "",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > differential diagnoses > updates comments 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "comments": "foo",
          "diagnosis": "",
        },
      },
      "name": "",
      "prescriptions": {
        "category": {
          "comments": "",
          "dose": "",
          "signature": "",
          "type": "",
        },
      },
      "proceduresAndServices": {
        "category": {
          "comments": "",
          "procedure": "",
          "when": "",
        },
      },
      "referrals": {
        "category": {
          "comments": "",
          "department": "",
          "when": "",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > differential diagnoses > updates diagnosis 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "code": "",
          "comments": "",
          "diagnosis": "foo",
        },
      },
      "name": "",
      "prescriptions": {
        "category": {
          "comments": "",
          "dose": "",
          "signature": "",
          "type": "",
        },
      },
      "proceduresAndServices": {
        "category": {
          "comments": "",
          "procedure": "",
          "when": "",
        },
      },
      "referrals": {
        "category": {
          "comments": "",
          "department": "",
          "when": "",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > general condition reducers > deletes initial condition 1`] = `
{
  "conditions": {},
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > general condition reducers > updates initial condition name 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "comments": "",
          "diagnosis": "",
        },
      },
      "name": "foo",
      "prescriptions": {
        "category": {
          "comments": "",
          "dose": "",
          "signature": "",
          "type": "",
        },
      },
      "proceduresAndServices": {
        "category": {
          "comments": "",
          "procedure": "",
          "when": "",
        },
      },
      "referrals": {
        "category": {
          "comments": "",
          "department": "",
          "when": "",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > prescriptions > deletes prescription 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "comments": "",
          "diagnosis": "",
        },
      },
      "name": "",
      "prescriptions": {},
      "proceduresAndServices": {
        "category": {
          "comments": "",
          "procedure": "",
          "when": "",
        },
      },
      "referrals": {
        "category": {
          "comments": "",
          "department": "",
          "when": "",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > prescriptions > updates prescription comments 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "comments": "",
          "diagnosis": "",
        },
      },
      "name": "",
      "prescriptions": {
        "category": {
          "comments": "foo",
          "dose": "",
          "signature": "",
          "type": "",
        },
      },
      "proceduresAndServices": {
        "category": {
          "comments": "",
          "procedure": "",
          "when": "",
        },
      },
      "referrals": {
        "category": {
          "comments": "",
          "department": "",
          "when": "",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > prescriptions > updates prescription dose 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "comments": "",
          "diagnosis": "",
        },
      },
      "name": "",
      "prescriptions": {
        "category": {
          "comments": "",
          "dose": "foo",
          "signature": "",
          "type": "",
        },
      },
      "proceduresAndServices": {
        "category": {
          "comments": "",
          "procedure": "",
          "when": "",
        },
      },
      "referrals": {
        "category": {
          "comments": "",
          "department": "",
          "when": "",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > prescriptions > updates prescription signature 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "comments": "",
          "diagnosis": "",
        },
      },
      "name": "",
      "prescriptions": {
        "category": {
          "comments": "",
          "dose": "",
          "signature": "foo",
          "type": "",
        },
      },
      "proceduresAndServices": {
        "category": {
          "comments": "",
          "procedure": "",
          "when": "",
        },
      },
      "referrals": {
        "category": {
          "comments": "",
          "department": "",
          "when": "",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > prescriptions > updates prescription type 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "comments": "",
          "diagnosis": "",
        },
      },
      "name": "",
      "prescriptions": {
        "category": {
          "comments": "",
          "dose": "",
          "signature": "",
          "type": "foo",
        },
      },
      "proceduresAndServices": {
        "category": {
          "comments": "",
          "procedure": "",
          "when": "",
        },
      },
      "referrals": {
        "category": {
          "comments": "",
          "department": "",
          "when": "",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > procedure or service > deletes procedure or services 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "comments": "",
          "diagnosis": "",
        },
      },
      "name": "",
      "prescriptions": {
        "category": {
          "comments": "",
          "dose": "",
          "signature": "",
          "type": "",
        },
      },
      "proceduresAndServices": {},
      "referrals": {
        "category": {
          "comments": "",
          "department": "",
          "when": "",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > procedure or service > updates comments 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "comments": "",
          "diagnosis": "",
        },
      },
      "name": "",
      "prescriptions": {
        "category": {
          "comments": "",
          "dose": "",
          "signature": "",
          "type": "",
        },
      },
      "proceduresAndServices": {
        "category": {
          "comments": "foo",
          "procedure": "",
          "when": "",
        },
      },
      "referrals": {
        "category": {
          "comments": "",
          "department": "",
          "when": "",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > procedure or service > updates procedure 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "comments": "",
          "diagnosis": "",
        },
      },
      "name": "",
      "prescriptions": {
        "category": {
          "comments": "",
          "dose": "",
          "signature": "",
          "type": "",
        },
      },
      "proceduresAndServices": {
        "category": {
          "comments": "",
          "procedure": "foo",
          "when": "",
        },
      },
      "referrals": {
        "category": {
          "comments": "",
          "department": "",
          "when": "",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > procedure or service > updates when 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "comments": "",
          "diagnosis": "",
        },
      },
      "name": "",
      "prescriptions": {
        "category": {
          "comments": "",
          "dose": "",
          "signature": "",
          "type": "",
        },
      },
      "proceduresAndServices": {
        "category": {
          "comments": "",
          "procedure": "",
          "when": "THIS_MONTH",
        },
      },
      "referrals": {
        "category": {
          "comments": "",
          "department": "",
          "when": "",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > referrals > deletes referral 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "comments": "",
          "diagnosis": "",
        },
      },
      "name": "",
      "prescriptions": {
        "category": {
          "comments": "",
          "dose": "",
          "signature": "",
          "type": "",
        },
      },
      "proceduresAndServices": {
        "category": {
          "comments": "",
          "procedure": "",
          "when": "",
        },
      },
      "referrals": {},
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > referrals > updates comments 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "comments": "",
          "diagnosis": "",
        },
      },
      "name": "",
      "prescriptions": {
        "category": {
          "comments": "",
          "dose": "",
          "signature": "",
          "type": "",
        },
      },
      "proceduresAndServices": {
        "category": {
          "comments": "",
          "procedure": "",
          "when": "",
        },
      },
      "referrals": {
        "category": {
          "comments": "foo",
          "department": "",
          "when": "",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > referrals > updates department 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "comments": "",
          "diagnosis": "",
        },
      },
      "name": "",
      "prescriptions": {
        "category": {
          "comments": "",
          "dose": "",
          "signature": "",
          "type": "",
        },
      },
      "proceduresAndServices": {
        "category": {
          "comments": "",
          "procedure": "",
          "when": "",
        },
      },
      "referrals": {
        "category": {
          "comments": "",
          "department": "foo",
          "when": "",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > referrals > updates when 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "comments": "",
          "diagnosis": "",
        },
      },
      "name": "",
      "prescriptions": {
        "category": {
          "comments": "",
          "dose": "",
          "signature": "",
          "type": "",
        },
      },
      "proceduresAndServices": {
        "category": {
          "comments": "",
          "procedure": "",
          "when": "",
        },
      },
      "referrals": {
        "category": {
          "comments": "",
          "department": "",
          "when": "THIS_MONTH",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > survey reducers > updates admit to hospital 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "comments": "",
          "diagnosis": "",
        },
      },
      "name": "",
      "prescriptions": {
        "category": {
          "comments": "",
          "dose": "",
          "signature": "",
          "type": "",
        },
      },
      "proceduresAndServices": {
        "category": {
          "comments": "",
          "procedure": "",
          "when": "",
        },
      },
      "referrals": {
        "category": {
          "comments": "",
          "department": "",
          "when": "",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "YES",
    "emergency": "",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > survey reducers > updates emergency 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "comments": "",
          "diagnosis": "",
        },
      },
      "name": "",
      "prescriptions": {
        "category": {
          "comments": "",
          "dose": "",
          "signature": "",
          "type": "",
        },
      },
      "proceduresAndServices": {
        "category": {
          "comments": "",
          "procedure": "",
          "when": "",
        },
      },
      "referrals": {
        "category": {
          "comments": "",
          "department": "",
          "when": "",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "NO",
    "sicknessLevel": 0,
  },
}
`;

exports[`plan reducers > survey reducers > updates sickness 1`] = `
{
  "conditions": {
    "condition": {
      "differentialDiagnoses": {
        "category": {
          "comments": "",
          "diagnosis": "",
        },
      },
      "name": "",
      "prescriptions": {
        "category": {
          "comments": "",
          "dose": "",
          "signature": "",
          "type": "",
        },
      },
      "proceduresAndServices": {
        "category": {
          "comments": "",
          "procedure": "",
          "when": "",
        },
      },
      "referrals": {
        "category": {
          "comments": "",
          "department": "",
          "when": "",
        },
      },
    },
  },
  "survey": {
    "admitToHospital": "",
    "emergency": "",
    "sicknessLevel": 5,
  },
}
`;
